JitRegCache: Cleanup

MerryMage opened the pull request:
First in a series of register cache PRs.

Encapsulate behavior appropriately to allow for future rework.

This PR has no behavioural changes.

BhaaLseN commented:
Nice cleanup there! `RegCache::SanityCheck` would probably be another candidate to return an `enum class` instead of those magic ints while we're at it.

Tilka commented on Source/Core/Core/PowerPC/Jit64/JitRegCache.cpp:
```diff
@@ -50,20 +50,15 @@ void RegCache::SetEmitter(XEmitter* emitter)
-  }
+  ASSERT_MSG(
+      DYNA_REC,
+      !std::any_of(m_xregs.begin(), m_xregs.end(), [](const auto& x) { return x.IsLocked(); }),
```
`!std::any_of` can be `std::none_of`

Tilka commented:
1. MakeDirty should probably not take an argument and instead be called conditionally.

2. Flushed and BoundTo sound like they test things rather than set them. Maybe rename them to SetFlushed and SetBoundTo?

Other than that, lgtm.

MerryMage commented:
Updated as above.

Updated `SanityCheck` to make more sense (i.e.: when invariants are violated), and [re-enabled logging of unflushed registers](https://github.com/dolphin-emu/dolphin/pull/7460/commits/54623c3ae8671c79871c68dcccde6b431f7db66e#diff-8f10a948b78d1ee44255543b4e83b1c2R941).

BhaaLseN commented:
You still have a `fixup!` commit in there, this should be rebased before it is merged.

MerryMage commented:
Rebased.
